kubetest2/pkg/process/exec.go

```diff
@@ -29,16 +29,20 @@ func Exec(argv0 string, args []string, env []string) error {
 	cmd := exec.Command(argv0, args...)
 	cmd.Env = env
 
-	// inherit all standard file descriptors, as if `syscall.Exec`ed
+	// inherit some standard file descriptors, as if `syscall.Exec`ed
 	cmd.Stdin = os.Stdin
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 
+	return execCmdWithSignals(cmd)
+}
+
+func execCmdWithSignals(cmd *exec.Cmd) error {
 	// setup listener to forward all signals
 	// TODO(bentheelder): what should this buffer size be?
 	signals := make(chan os.Signal, 5)
 	signal.Notify(signals)
-	defer close(signals)
+	defer signal.Stop(signals)
```

Comment:
This is the bug I fixed:

> Stop causes package signal to stop relaying incoming signals to c. It undoes the effect of all prior calls to Notify using c. When Stop returns, it is guaranteed that c will receive no more signals.

This is the correct thing to do for signal channels, not closing the channel.